### PR TITLE
chore: remove refs to AllowUpdateToExpired/Frozen client booleans

### LINKED ISSUE #1237: Remove AllowUpdateAfterExpiry and AllowUpdateAfterMisbehaviour references
testing/config.go fields can be removed -- this is an API breaking change

### PR DESCRIPTION
**Pull request by charleenfei**
## Description

closes #1237

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Comment by charleenfei** [comment]
question  -- should the proto fields also be removed? currently it's API breaking (still need to finish updating the CHANGELOG) but updated the protos would make it also state machine breaking (right?) should we just do this in one go?

File: CHANGELOG.md [hunk 1]
```diff
@@ -54,6 +54,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 * (transfer)[\#1565](https://github.com/cosmos/ibc-go/pull/1565) Removing `NewErrorAcknowledgement` in favour of `channeltypes.NewErrorAcknowledgement`.
 * (channel)[\#1565](https://github.com/cosmos/ibc-go/pull/1565) Updating `NewErrorAcknowledgement` to accept an error instead of a string and removing the possibility of non-deterministic writes to application state.
 * (core/04-channel)[\#1636](https://github.com/cosmos/ibc-go/pull/1636) Removing `SplitChannelVersion` and `MergeChannelVersions` functions since they are not used.
+* (light-clients/tendermint)[\#1768](https://github.com/cosmos/ibc-go/pull/1768/files) Removed `AllowUpdateAfter...` booleans as they are deprecated (see ADR026)
```
Comment (chatton) [inline CHANGELOG.md:57]: [nit] maybe we should explicitly list the booleans to make it more clear what exactly was removed?

**Review by chatton — APPROVED** [review]
LGTM! Just one small comment on the changelog entry 🥇

File: modules/light-clients/07-tendermint/types/upgrade.go [hunk 2]
```diff
@@ -101,7 +101,6 @@ func (cs ClientState) VerifyUpgradeAndUpdateState(
 	newClientState := NewClientState(
 		tmUpgradeClient.ChainId, cs.TrustLevel, cs.TrustingPeriod, tmUpgradeClient.UnbondingPeriod,
 		cs.MaxClockDrift, tmUpgradeClient.LatestHeight, tmUpgradeClient.ProofSpecs, tmUpgradeClient.UpgradePath,
-		cs.AllowUpdateAfterExpiry, cs.AllowUpdateAfterMisbehaviour,
```
Comment (colin-axner) [inline modules/light-clients/07-tendermint/types/upgrade.go:104]: I guess there's a sideaffect here. These parameters will be set to false, false upon upgrades (even if the previous values were true, true). Shouldn't cause an issue, but is worth being aware of

**Review by colin-axner — APPROVED** [review]
LGTM, nice work!

**Comment by colin-axner** [comment]
It probably makes sense to include this in v6 instead of v5 (ie wait for the release branch to be created), just in case there's any issues for users with the changing of `NewClientState` API (thinking of backwards compatibility to older versions)

File: CHANGELOG.md (hunk 1 above)
Comment (charleenfei) [reply CHANGELOG.md:57]: makes sense, i will update!

File: modules/light-clients/07-tendermint/types/upgrade.go (hunk 2 above)
Comment (charleenfei) [reply modules/light-clients/07-tendermint/types/upgrade.go:104]: i can add an inline doc for this :)

**Comment by charleenfei** [comment]
> It probably makes sense to include this in v6 instead of v5 (ie wait for the release branch to be created), just in case there's any issues for users with the changing of `NewClientState` API (thinking of backwards compatibility to older versions)

cc @crodriguezvega

File: modules/light-clients/07-tendermint/types/upgrade.go (hunk 2 above)
Comment (charleenfei) [reply modules/light-clients/07-tendermint/types/upgrade.go:104]: added a note in upgrade docs about this

**Pull request merged**